### ISSUE TITLE
CuSolver: Switch to 64 bit api to allow for eigh on matrices > than 26732x26732

### ISSUE THREAD

**Issue by PhilipVinc** [jax-ml/jax#23413]
Jaxlib links to CuSolver 32 bit api, which has a hard limit on workspace size which makes it such that it is not possible to diagonalise matrices larger than 26k^2 when using np.float64.

**Comment by norabelrose** [comment]
I'm running into a similar issue with `jax.linalg.svd`

**Comment by PhilipVinc** [comment]
Yeah, this affects all CuSolver apis, so svd and various factorizations as well…

**Comment by dfm** [comment]
Great suggestion! I'm in the midst of updating all the cuSolver wrappers so I'll plan on getting this in as part of that process. I'd guess that I probably won't be able to land this before the next JAX release, but I'll try!

For reference, it looks like there's an open issue suggesting this for the CPU backend too: https://github.com/google/jax/issues/20904

**Comment by norabelrose** [comment]
Thanks a lot for this @dfm, even just a PR / branch that uses the 64 bit api would be very useful since I am trying to run SVD on some large matrices for a project right now.

**Comment by dfm** [comment]
Sure - I can prioritize SVD. Just to confirm, you're running on a GPU, @norabelrose?

**Comment by norabelrose** [comment]
> Sure - I can prioritize SVD. Just to confirm, you're running on a GPU, @norabelrose?

Yep, that's right. Thanks!

**Comment by PhilipVinc** [comment]
I think this has now been solved
